test/validator.go

```diff
@@ -1,197 +1,273 @@
-		Threshold: threshold,
-		PubKeys:   make(map[string]cometcrypto.PubKey),
-	}
-	if err := testValidator.genPrivKeyAndShards(nil, chainIDs...); err != nil {
-		return nil, err
-	}
-	return testValidator, nil
+	signerPort       = "2222"
+	signerPortDocker = signerPort + "/tcp"
+
+	signerImage        = "horcrux-test"
+	binary             = "horcrux"
+	signerImageUidGid  = "2345:2345"
+	signerImageHomeDir = "/home/horcrux"
+)
+
+// chainWrapper holds the initial configuration for a chain to start from genesis.
+type chainWrapper struct {
+	chain           **cosmos.CosmosChain
+	totalValidators int // total number of validators on chain at genesis
+	totalSentries   int // number of additional sentry nodes
+	modifyGenesis   func(cc ibc.ChainConfig, b []byte) ([]byte, error)
+	preGenesis      func(ibc.ChainConfig) error
 }
 
-func NewHorcruxValidatorWithPrivValKey(
-	tl Logger,
-	pool *dockertest.Pool,
-	networkID string,
-	home string,
-	index int,
-	numSentries int,
-	numSigners int,
-	threshold uint8,
-	chainType *ChainType,
-	privValKey privval.FilePVKey,
-) (*Validator, error) {
-	chainID := chainType.ChainID
-	sentries := make(map[string]Nodes)
-	sentries[chainID] = MakeNodes(index, numSentries, home, chainID, chainType, pool, networkID, tl)
-	testValidator := &Validator{
-		Index:     index,
-		Sentries:  sentries,
-		Signers:   MakeSigners(index, numSigners, home, pool, networkID, tl),
-		tl:        tl,
-		Home:      home,
-		Threshold: threshold,
-		PubKeys:   make(map[string]cometcrypto.PubKey),
+// startChains starts the given chains locally within docker composed of containers.
+func startChains(
+	ctx context.Context,
+	t *testing.T,
+	logger *zap.Logger,
+	client *client.Client,
+	network string,
+	chains ...chainWrapper,
+) {
+	err := BuildHorcruxImage(ctx, client)
+	require.NoError(t, err)
+
+	cs := make([]*interchaintest.ChainSpec, len(chains))
+	for i, c := range chains {
+		cs[i] = &interchaintest.ChainSpec{
+			Name:          testChain,
+			Version:       testChainVersion,
+			NumValidators: &c.totalValidators,
+			NumFullNodes:  &c.totalSentries,
+			ChainConfig: ibc.ChainConfig{
+				ModifyGenesis: c.modifyGenesis,
+				PreGenesis:    c.preGenesis,
+			},
+		}
 	}
-	if err := testValidator.genPrivKeyAndShards(&privValKey, chainID); err != nil {
-		return nil, err
+
+	cf := interchaintest.NewBuiltinChainFactory(logger, cs)
+
+	cfChains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+
+	ic := interchaintest.NewInterchain()
+
+	for i, c := range cfChains {
+		chain := c.(*cosmos.CosmosChain)
+		*chains[i].chain = chain
+		ic.AddChain(chain)
 	}
-	return testValidator, nil
-}
 
-// Name is the name of the test validator
-func (tv *Validator) Name() string {
-	return fmt.Sprintf("validator-%d-%s", tv.Index, tv.tl.Name())
-}
+	err = ic.Build(ctx, nil, interchaintest.InterchainBuildOptions{
+		TestName:  t.Name(),
+		Client:    client,
+		NetworkID: network,
+		// BlockDatabaseFile: interchaintest.DefaultBlockDatabaseFilepath(),
```

Comment:
nit: Can you add a comment informing a reader why we're commenting this out?

Reply:
we won't ever need this in horcrux I don't think, so I removed it.